### ISSUE TITLE
plotOptions.scatter.marker.states.select.lineColor not operating per API for null value

### ISSUE THREAD

**Issue by caguthrie** [highcharts/highcharts#4393]
http://api.highcharts.com/highcharts#plotOptions.scatter.marker.states.select

For explicit null values, lineColor is supposed to be the series color when selected.  Instead, it is white.

See this fiddle from the API (change to null):
http://jsfiddle.net/gh/get/jquery/1.7.2/highslide-software/highcharts.com/tree/master/samples/highcharts/plotoptions/series-marker-states-select-linecolor/

**Comment by erfangc** [comment]
for anyone else coming across this issue - take a look at series.scatter.marker.states.select (instead of the one under plotOptions), you will need to figure out what color the series take on and act accordingly.

**Comment by pawelfus** [comment]
Workaround: set `lineColor: null` for marker. Demo: http://jsfiddle.net/9n8kpLjf/

**Comment by pawelfus** [comment]
It never worked the way described in the docs. I'm not sure if this is a bug in docs, or in the code.

Internal notes:

This line is the culprit [Series.js](https://github.com/highslide-software/highcharts.com/blob/master/js/parts/Series.js#L1103) - where `stateOptions.select.fillColor` is set to null, where `series.convertAttribs()` is using `pick` [here](https://github.com/highslide-software/highcharts.com/blob/master/js/parts/Series.js#L1037) which of course removes nullified options.

**Comment by pawelfus** [comment]
It never worked the way described in the docs. I'm not sure if this is a bug in docs, or in the code.

Internal notes:

This line is the culprit in [Series.js](https://github.com/highslide-software/highcharts.com/blob/master/js/parts/Series.js#L1103) - where `stateOptions.select.fillColor` is set to `null`, and since `series.convertAttribs()` is using `pick` [here](https://github.com/highslide-software/highcharts.com/blob/master/js/parts/Series.js#L1037) which of course removes nullified options  (or rather is using next-base option instead).

In other words we can have that priority for states:

> Conversion: lineColor -> stroke: (marker.states.lineColor) `null` -> (marker.lineColor) `#FFFFFF` -> (series.color) `#7cb5ec`

**Comment by caguthrie** [comment]
Looks like the line in the docs "When null, the series' or point's color is used. Defaults to #000000." might need to be moved from `lineColor` to `fillColor`.  That would solve the continuity between docs and code.
